eng/scripts/Automation-Sdk-Generate.ps1

```diff
@@ -25,29 +25,24 @@ Write-Host "Updating autorest.md files for all the changed swaggers."
 $sdksInfo = @{}
 $headSha = $inputFileContent.headSha
 $repoHttpsUrl = $inputFileContent.repoHttpsUrl
+$regexToFindSha = "https:\/\/[^`"]*[\/][0-9a-f]{4,40}[\/]"
 foreach ($path in $autorestFilesPath) {
   $fileContent = Get-Content $path
   foreach ($inputFilePath in $inputFilePaths) {
-    $isUpdatedLines = $false
     $escapedInputFilePath = [System.Text.RegularExpressions.Regex]::Escape($inputFilePath)
-    $regexForMatchingShaAndPath = "https:\/\/[^`"]*[\/][0-9a-f]{4,40}[\/]$escapedInputFilePath"
-    $updatedLines = foreach ($line in $fileContent) {
+    $regexForMatchingShaAndPath = $regexToFindSha + $escapedInputFilePath
+
+    foreach ($line in $fileContent) {
       if ($line -match $regexForMatchingShaAndPath) {
-        $line -replace $regexForMatchingShaAndPath, "$repoHttpsUrl/blob/$headSha/$inputFilePath"
+        $fileContent -replace $regexToFindSha, "$repoHttpsUrl/blob/$headSha/" | Set-Content -Path $path
 
-        $isUpdatedLines = $true
         $sdkpath = (get-item $path).Directory.Parent.FullName | Resolve-Path -Relative
         if (!$sdksInfo.ContainsKey($sdkpath)) {
           $specReadmePath = $inputFileContent.relatedReadmeMdFiles -match $inputFilePath
           $sdksInfo.Add($sdkpath, $specReadmePath)
         }
+        break
```

Comment:
Why do we want to break here?

Reply:
If we found any match with the changed file in PR in an autorest.md, then [here](https://github.com/Azure/azure-sdk-for-net/blob/ee20c1adfdcc63342626f09153866a6c48d24bf0/eng/scripts/Automation-Sdk-Generate.ps1#L37) I'm updating all the sha in an autorest.md and added the `break` to get out of the loop. And then it will check the next autorest.md file in the outer loop to do the same.